packages/block-editor/src/components/colors-gradients/style.scss

```
@@ -77,18 +77,33 @@ $swatch-gap: 12px;
 	border-right: 1px solid $gray-300;
 	border-bottom: 1px solid $gray-300;
 
-	&.first {
+	// Identify the first instance as the item that is not a subsequent item to an
+	// item with the same class. The `~` selector is used instead of `+` to take
+	// into account any potential `ToolsPanelItem` placeholder item.
+	&:not(& ~ &) {
```

```suggestion
	// Identify the first visible instance as placeholder items will not have this class.
	&:nth-child(1 of &) {
```

With the change below avoiding the need for `:has`, and fixing Firefox, perhaps this selector is clearer and doesn't require the lengthy explanation. (See [coverage](https://caniuse.com/css-nth-child-of) for selector list in nth-child)

---

TIL about `of` selector list argument 👍